aws-auth-cognito/src/main/java/com/amplifyframework/statemachine/codegen/data/AuthConfiguration.kt

```diff
@@ -64,5 +110,24 @@ internal data class AuthConfiguration internal constructor(
             else
                 AuthFlowType.USER_SRP_AUTH
         }
+
+        private fun getPasswordProtectionSettings(authConfig: JSONObject?): PasswordProtectionSettings? {
+            val passwordSettings = authConfig?.optJSONObject("passwordProtectionSettings") ?: return null
+            val passwordLength = passwordSettings.optInt("passwordPolicyMinLength")
+            val passwordRequirements = passwordSettings.optJSONArray("passwordPolicyCharacters")?.map {
+                getString(it)
+            } ?: emptyList()
+            return PasswordProtectionSettings(
+                length = passwordLength,
```

Comment:
what happens on `passwordSettings.optInt("passwordPolicyMinLength")`  when the JSONObject does not contain the key-value? Does it return null for `passwordLength`, if so what happens when creating the PasswordProtectionSettings's `length` with null?

Reply:
`optInt` returns 0 if the field is missing or is not a number